### PR TITLE
Changed configs to use engines by key

### PR DESCRIPTION
**Pull request by bilfr3d**
Instead of a singular "database" key specifying database configuration, all the engine types have their own individual key so users don't have confusion around editing the configuration.

### CODE REVIEW

File: testdata/configs/session.json [hunk 1]
```diff
@@ -9,12 +9,12 @@
...
-	"database": {
+	"mongodb": {
 		"url": "mongodb://localhost/MongoDatabase-Session"
 	},
 	"redis": {
 		"host": "localhost",
 		"post": 6379,
-		"password": "optional, may leave blank if your redis has no password"
+		"password": ""
 	}
```
Comment (fatchan) [inline testdata/configs/session.json]: maybe you should nest these in a "storageEngines" key

File: server.js [hunk 2]
```diff
@@ -171,9 +182,14 @@ module.exports = async (baseDir, configsDirectory, routesDirectory)=>{
...
 				case "redis":
-					Store = new (require('connect-redis')(session))({
-						client: new (require('ioredis'))(config.session.redis)
-					});
+					if(config.session.redis && config.session.redis.host && config.session.redis.port) {
+						Store = new (require('connect-redis')(session))({
+							client: new (require('ioredis'))(config.session.redis)
+						});
+					} else {
+						logErr("Incorrect session database configuration options. Check your config and try again.");
```
Comment (fatchan) [inline server.js]: maybe include the name of invalid input and config file location here

File: testdata/configs/session.json (hunk 1 above)
Comment (bilfr3d) [reply testdata/configs/session.json]: Although that would make scaling better, objects would have to be retrieved with `config.session.storageEngines[config.session.storageEngine]`

Comment (fatchan) [reply testdata/configs/session.json]:
```js
const { storageEngines, storageEngine } = config.sessions;
storageEngines[storageEngine];
```
if u dont want a long line, but it doesnt make much diff

**Pull request merged**